frame/contracts/src/exec.rs

```diff
@@ -1046,8 +1056,14 @@ where
 			return Ok(())
 		}
 
+		// If the caller is root, we don't need to transfer.
+		let caller = match self.caller() {
+			Caller::Account(caller) => caller,
+			Caller::Root => return Ok(()),
+		};
```

Comment:
I actually want that particular code to be explicit about only returning `Ok` when the Caller is root, instead of returning `Ok` when the caller isn't an Account. This could avoid future issues if there are more Callers other than `Account` and `Root`